### PR TITLE
Update Red Hat OpenShift support statement

### PR DESCRIPTION
**Pull request by conceptualshark**
## Description

From feedback on the OpenShift support statement: "does this documentation imply that for new installations we will no longer recommend OpenShift?" Cleaning this up to make it more explicit what is and is not supported. This uses the life cycle phases defined by Red Hat on top of the provided table to provide more specific information on support.

Original suggestion: "We support only the OpenShift versions that are currently supported by Red Hat—that is, versions within their End of Support Date. For more details, please refer to the Red Hat OpenShift Container Platform Life Cycle Policy." 

Internal slack: https://camunda.slack.com/archives/C01Q05D6YUU/p1731998800535919

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Review by aabouzaid — APPROVED** [review]

**Comment by conceptualshark** [comment]
adding @camunda/tech-writers to get some additional feedback on the wording here, I'd like to get this merged if it seems to be the right direction for clarifying the support window.

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ We conduct testing and ensure compatibility against the following OpenShift vers
 | 4.13.x            | November 17, 2024                                                                 |
 
 :::caution
-Compatibility is not guaranteed for OpenShift versions no longer supported by Red Hat, as per the End of Support Date. For more information, refer to the [Red Hat OpenShift Container Platform Life Cycle Policy](https://access.redhat.com/support/policy/updates/openshift).
+Camunda supports OpenShift versions that fall within the Red Hat General Availability, Full Support, and Maintenance Support life cycle phases. For more information, refer to the [Red Hat OpenShift Container Platform Life Cycle Policy](https://access.redhat.com/support/policy/updates/openshift).
```
Comment (mesellings) [inline docs/self-managed/setup/deploy/openshift/redhat-openshift.md:36]:
```suggestion
Camunda 8 only supports OpenShift versions in Red Hat General Availability, Full Support, and Maintenance Support life cycle phases. For more information, refer to the [Red Hat OpenShift Container Platform Life Cycle Policy](https://access.redhat.com/support/policy/updates/openshift).
```
Wording lgtm, just an idea for rewording here but feel free to ignore as I'm not sure it's any better - I did add Camunda 8 in however as I wasn't sure if we should clarify whether we mean Camunda (the company) supports..., or Camunda 8 (the software) supports...

**Review by mesellings — APPROVED** [review]
Lgtm - only one non-blocking suggestion 👍

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md (hunk 1 above)
Comment (conceptualshark) [reply docs/self-managed/setup/deploy/openshift/redhat-openshift.md]: I like these suggestions - I've gone ahead and added them in, thank you!

**Review by mesellings — APPROVED** [review]
Lgtm 👍 🚀

**Pull request merged**